### DISCUSSION TITLE
Expected behaviour when editing a listing with an existing null return_policy_id after Jan 31st?

### DISCUSSION THREAD

**Discussion by PMStanley** [etsy/open-api#827] [Q&A] [answered]
The breaking change due for Jan 31st states (emphasis mine):

> Return policies will be required when creating or **editing** a listing. Note: this particular requirement does not apply to listings of EU-based shops.

I've just been checking how the API responds when editing a draft listing, created on the website, with no return policy set and I'm not sure what the expected action is here from the notes?

Requesting the listing with getListing shows that return_policy_id is null, attempting to update the listing with return_policy_id set to nothing, null or 0 results in an error each time.
Is the expectation that we continue to update listings without a return policy set post Jan 31st *without sending return_policy_id in the request* or do we have to set one regardless of whether the update would change that part of the listing (e.g. a description update)?

What about listings where return_policy_id is already set, do we need to send the existing return_policy_id with each update? Or will it work as it currently does where we don't need to send that field because it's already set?

**Comment by Makebuildmodify** [comment]
I've been setting my new listings' return policies when creating a draft listing without any trouble. Although, I haven't tried to update an active listing from no policy to having a policy.

**Reply by PMStanley** [reply]
My use cases don't update the policy so I'm keen to understand what I should be doing, if anything, when editing listings after Jan 31st.

Despite the API docs saying that return_policy_id is nullable there doesn't appear to be any way to actually set it to null so my question to @nateswart is what is the expected action on a draft listing being edited with no policy assigned using updateListing? 

Do we just ignore return_policy_id? Will the update fail if we do that? If we do need to send return_policy_id how do we actually set it to null?
Is the expectation that if a draft listing doesn't have a policy and we update some part of it, say the description, we *have* to force our users to assign a policy whether they want to or not?

What about listings where return_policy_id is already set, do we need to echo the value back when calling updateListing or can we safely ignore it.

I don't have access to a US shop to test, breaking changes that only apply to specific regions are inherently difficult to test so we need clear documentation and examples.

**Reply by Makebuildmodify** [reply]
Etsy provides a generic 14 day return policy. So that's the one I used to satisfy the requirement for my draft listings. I figured that I could bulk edit them later if I need to. I know that's not what you wanted to hear but it's a way to get by. It would be nice to have control over that choice for the User.

**Reply by PMStanley** [reply]
Appreciate your input. That's my backup plan for editing draft listings without a policy assigned but doesn't cover the scenario where a policy is already assigned. Can we drop return_policy_id from the request or do we need to echo back what the listing is already set to?

We really need breaking changes to be well documented and the API to act as described rather than guessing what we should do and finding out that fields marked as nullable aren't actually nullable.

**Reply by Makebuildmodify** [reply]
I agree, the requirements on this change are pretty ambiguous.

I would imagine if you use getShopReturnPolicy that you wouldn't be changing the state of the listing. So, you could test that way. 

Also, the EU law regarding returns requires that some shops offer refunds and returns. So, (wild guess) I'm wondering if Etsy will be automatically setting the minimum required return policy for EU shops. This would explain why it's not necessary to add it during listing creation. IDK?

**Comment by PMStanley** [comment, marked as answer]
Big thanks to Artemis for coming back with the following answers:

> Question 1: 
> What is the expected action when updating only the description of a draft listing using the ‘updateListing’ endpoint?
> Answer: 
> You do not need to send the return_policy_id if you are not changing the state of the listing to active/publishing the listing. The field is only required if the listings does not have a return policy and you are trying to publish. 
> 
> Question 2: 
> Can we ignore sending this parameter as it’s already set on the listing?
>  Answer:
> Correct. You do not need to send the return_policy_id parameter if the listing already has a return policy.
